### PR TITLE
[EPROD-919]: Better estimate fee amount in BftBridge contract

### PR DESCRIPTION
**Pull request by itsyaasir**
- Refactor gas fee calculation to use a 15% buffer
- Simplify the gas fee calculation logic
- Format with forge configuration

## Issue ticket

Issue ticket link:

### Code conventions

- [x] I have performed a self-review of my code
- [x] Every new function is documented
- [x] Object names are auto explicative
- [x] The PR follows the [project conventions](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/23330839/Conventions)
- [x] The code follows the [style guidelines of the project](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/24444929/Rust+Conventions)
- [x] The code follows the [project security best practices](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/35225620/Security+Considerations)

### Security 

- [x] The PR does not break APIs backward compatibility
- [x] The PR does not break the stable storage backward compatibility

### Testing 

- [x] Every function is properly unit tested
- [x] I have added integration tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

**Review by F3kilo — CHANGES_REQUESTED** [review]
Can we add a test to check if minter canister native tokens balance doesn't decrease after several bridge operations?

File: solidity/src/BftBridge.sol
Comment (ufoscout) [inline solidity/src/BftBridge.sol:1]: remove the `additionalGasFee` constant

File: solidity/src/BftBridge.sol [hunk 1]
```diff
@@ -181,8 +181,10 @@ contract BFTBridge is TokenManager, UUPSUpgradeable, OwnableUpgradeable, Pausabl
             order.feePayer != address(0) && msg.sender == minterCanisterAddress
                 && address(feeChargeContract) != address(0)
         ) {
-            uint256 gasFee = initGasLeft - gasleft() + additionalGasFee;
-            uint256 fee = gasFee * tx.gasprice;
+            uint256 additionalGasConsumed = initGasLeft - gasleft();
+            uint256 additionalGasFeeConsumed = (additionalGasConsumed * 115) / 100; // Add 15% buffer
+            uint256 fee = additionalGasFeeConsumed * tx.gasprice;
```
Comment (ufoscout) [inline solidity/src/BftBridge.sol:186]: can you add a test to check if the fee is the expected one?

**Review by ufoscout — CHANGES_REQUESTED** [review]

File: solidity/src/BftBridge.sol (hunk 1 above)
Comment (F3kilo) [reply solidity/src/BftBridge.sol]: The code has been re-worked

File: solidity/src/BftBridge.sol
Comment (F3kilo) [reply solidity/src/BftBridge.sol:1]: Now it is used in the fee calculation logic.

**Review by veeso — APPROVED** [review]

**Review by F3kilo — APPROVED** [review]

**Review by ufoscout — APPROVED** [review]

**Pull request merged**